tribe_loader.py

```diff
@@ -177,48 +192,174 @@ def load_to_tribe(main_config_file, geneset_info, prefer_update=False):
+                    logger.warning('No Entrez IDs were found for gene %s '
+                                   'and cross-reference DB %s.',
+                                   gene, geneset_info['xrdb'])
 
             if pub_set_annotations != old_annotations:
                 # Annotations have changed
                 geneset_info['geneset'] = gs_response['resource_uri']
                 geneset_info['parent'] = gs_response['tip']['resource_uri']
                 geneset_info['description'] = 'Updating annotations.'
 
+                logger.info('Creating new version for geneset %s',
+                            geneset_info['title'])
                 response = create_remote_version(access_token, geneset_info,
                                                  tribe_url)
             else:
                 # Do not create a new version or geneset
                 logger.info('Geneset with title %s already exists with the '
                             'same annotations in Tribe. Not attempting to '
-                            'save this geneset.', geneset_info['title'])
+                            'save a new geneset nor geneset version.',
+                            geneset_info['title'])
                 response = {}
                 response['content'] = (
                     'There is already a geneset with the slug "{0}" and '
-                    'annotations {1} saved in Tribe. A new geneset has not '
-                    'been saved.').format(geneset_info['slug'],
-                                          geneset_info['annotations'])
+                    'annotations {1} saved in Tribe. Neither a geneset nor '
+                    'a geneset version have been created.').format(
+                        geneset_info['slug'], geneset_info['annotations'])
                 response['status_code'] = 409
         else:
             # No geneset with this geneset 'slug' and creator username exists
             # yet, so create it
+            logger.info('Creating geneset %s', geneset_info['title'])
             response = create_remote_geneset(access_token, geneset_info,
                                              tribe_url)
 
     else:
+        logger.info('Creating geneset %s', geneset_info['title'])
         response = create_remote_geneset(access_token, geneset_info,
                                          tribe_url)
     return response
 
 
-def write_json_file(geneset_info, json_filename):
-    with open(json_filename, "w") as outfile:
-        json.dump(geneset_info, outfile, indent=2)
+def compare_genesets(tribe_genesets, processed_genesets):
+    tribe_geneset_dict = {}
+    proc_geneset_dict = {}
+
+    changed_genesets = []
+
+    # We will use the gene set 'slugs' as the identifiers for each gene set
+    for gs in tribe_genesets:
+        tribe_geneset_dict[gs['slug']] = gs
+
+    for gs in processed_genesets:
+        proc_geneset_dict[gs['slug']] = gs
+
+    changed_ctr = 0
+    unchanged_ctr = 0
```

Comment:
Is it necessary to keep track of unchanged items? Isn't it equal to the total number of items minus `changed_ctr`?

Reply:
Ah yes, and actually I don't need the `changed_ctr` either! That was just a slip in my mind - I can just get it with `len(changed_genesets)`, so I will eliminate both of these counters!